src/services/symbolDisplay.ts

```diff
@@ -638,26 +648,50 @@ namespace ts.SymbolDisplay {
             displayParts.push(spacePart());
         }
 
-        function addFullSymbolName(symbolToDisplay: Symbol, enclosingDeclaration?: Node) {
+        function addFullSymbolName(symbolToDisplay: Symbol, enclosingDeclaration?: Node, indexInfos?:IndexInfo[]) {
             if (alias && symbolToDisplay === symbol) {
                 symbolToDisplay = alias;
             }
-            const fullSymbolDisplayParts = symbolToDisplayParts(typeChecker, symbolToDisplay, enclosingDeclaration || sourceFile, /*meaning*/ undefined,
-                SymbolFormatFlags.WriteTypeParametersOrArguments | SymbolFormatFlags.UseOnlyExternalAliasing | SymbolFormatFlags.AllowAnyNodeKind);
+            let fullSymbolDisplayParts = symbolToDisplayParts(typeChecker, symbolToDisplay, enclosingDeclaration || sourceFile, /*meaning*/ undefined,
+            SymbolFormatFlags.WriteTypeParametersOrArguments | SymbolFormatFlags.UseOnlyExternalAliasing | SymbolFormatFlags.AllowAnyNodeKind);
+            if(symbolToDisplay.flags & SymbolFlags.Signature){
+                if(indexInfos){
+                    let index = 1;
+                    fullSymbolDisplayParts[index++] = punctuationPart(SyntaxKind.OpenBracketToken);
+                    if(length(indexInfos)){
+                        //Needed to handle more than one type of index
+                        for(let info=0; info<indexInfos.length; info++){
+                            const indexTypeDisplayParts =  typeToDisplayParts(typeChecker, indexInfos[info].keyType);
+                            //Needed to handle template literals
+                            for (const part of indexTypeDisplayParts){
+                                fullSymbolDisplayParts[index++] = part; 
+                            }
+                            if(info !== indexInfos.length-1){
+                                fullSymbolDisplayParts[index++] = spacePart();
+                                fullSymbolDisplayParts[index++] = punctuationPart(SyntaxKind.BarToken);
+                                fullSymbolDisplayParts[index++] = spacePart();
+                            }
+                        }
+                        fullSymbolDisplayParts[index] = punctuationPart(SyntaxKind.CloseBracketToken);
+                    }
+                }
+                else{
+                    fullSymbolDisplayParts[2] = textOrKeywordPart(TypeFlags[`1`]); //This is the fallback in case else fails

```

Comment:
...? `TypeFlags[1]` doesn't sound like any keyword I know. What's it supposed to be referring to?

Reply:
returns `Any`

Reply:
Should probably manually write out `"Any"` then.